src/Conversion/ONNXToTOSA/Tensor/Split.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------- Split.cpp - Split Op---------===//
+//
+// Copyright (c) 2023 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX SplitOp operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTOSA/DialectBuilder.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+
+using namespace mlir;
+namespace onnx_mlir {
+namespace {
+class ONNXSplitOpLoweringToTOSA : public OpConversionPattern<ONNXSplitOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXSplitOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXSplitOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    Value input = adaptor.getInput();
+    if (!onnx_mlir::hasShapeAndRank(input)) {
+      return rewriter.notifyMatchFailure(
+          op, "input is not a ranked shaped tensor");
+    }
+
+    ShapedType inputType = input.getType().cast<ShapedType>();
```

Comment:
member cast methods are deprecated
```suggestion
    ShapedType inputType = cast<ShapedType>(input.getType);
```